src/scripts/match-mondo-sources-all-lexical.py

```diff
@@ -109,18 +112,25 @@ def run(input: str, config: str, rules: str, rejects: str, output: str):
     # )
 
     prefix_of_interest = yml["subject_prefixes"]
-
     resource = OntologyResource(slug=f"sqlite:///{Path(input).absolute()}")
     oi = SqlImplementation(resource=resource)
     ruleset = load_mapping_rules(rules)
     # syn_rules = [x.synonymizer for x in ruleset.rules if x.synonymizer]
-    lexical_index = create_lexical_index(oi=oi, mapping_rule_collection=ruleset)
-    save_lexical_index(lexical_index, OUT_INDEX_DB)
+    # TODO: uncomment
+    # lexical_index = create_lexical_index(oi=oi, mapping_rule_collection=ruleset)
+    # save_lexical_index(lexical_index, OUT_INDEX_DB)
+    # TODO: /uncomment
+
+    # TODO temp delete after
+    import pickle
+    pp = '/Users/joeflack4/Desktop/lexical_index.pickle'
+    # pickle.dump(lexical_index, open(pp, "wb"))
+    lexical_index = pickle.load(open(pp, "rb"))
```

Comment:
## Caching `.pickle` > `.yml` for `tmp/merged.db.lexical.yaml`? @matentzn @hrshdhgd 
@hrshdhgd I don't think will be able to review this but @matentzn if you could consider my suggestions.

<details><summary>Details</summary>
<p>

### Add caching
For development, caching helps. These two lines for `create_lexical_index()` and `save_lexical_index()` are taking 20 minutes. 

### Cache `.pickle` > `.yml`?
The `tmp/merged.db.lexical.yaml` currently being saved is 266MB, and I don't see it being used anywhere. Maybe it is for manual inspection / troubleshooting purposes?

Also, it takes a long time to save. I'm not sure how long. The way I executed it, PyCharm crashed.

If we add caching, I would suggest pickle. It both saved and loaded quickly, and was only 750KB. I was shocked at the size difference from the 266MB `yml`. I wonder if there was some lossiness in the pickling, but I don't yet know how to even check for that. Either way, I've been running this and reloading from pickle, and it has not seemed to cause any issues. The script still continues to run for another 45 minutes.

</p>
</details>

Reply:
Maybe make a separate issue about this and discuss with Trish and Harshad when you see them?